python/pyspark/serializers.py

```diff
@@ -343,6 +343,8 @@ def _load_stream_without_unbatching(self, stream):
         key_batch_stream = self.key_ser._load_stream_without_unbatching(stream)
         val_batch_stream = self.val_ser._load_stream_without_unbatching(stream)
         for (key_batch, val_batch) in zip(key_batch_stream, val_batch_stream):
+            key_batch = key_batch if hasattr(key_batch, '__len__') else list(key_batch)
```

Comment:
Could we add a small comment that this is required because `_load_stream_without_unbatching` could return an  iterator of iterators in this case?